### PR TITLE
Dcc show publish by

### PR DESCRIPTION
**Pull request by xyfy**
project list

project detail

#559

### CODE REVIEW

File: src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Dcc.Service.Admin.Application.App.Queries
+{
+    public record LatestReleaseQueryByApp(List<int> AppIds, int? EnvClusterId = null) : Query<List<LatestReleaseConfigModel>>
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs:6]: LatestReleaseQueryByAppQuery

File: src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseByProjectQuery.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Dcc.Service.Admin.Application.App.Queries;
+
+public record LatestReleaseByProjectQuery(List<BuildingBlocks.StackSdks.Pm.Model.ProjectModel> Projects, int? EnvClusterId = null) : Query<List<LatestReleaseConfigModel>>
+{
+    public override List<LatestReleaseConfigModel> Result { get; set; }
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseByProjectQuery.cs:8]: = new();

File: src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs (hunk 1 above)
Comment (xyfy) [reply src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs:6]: LatestReleaseByAppQuery？

File: src/Services/Masa.Dcc.Service/Application/App/QueryHandler.cs [hunk 3]
```diff
@@ -57,6 +58,46 @@ public async Task GetPublicConfigsAsync(PublicConfigsQuery query)
...
+        public async Task GetLatestReleaseConfigByProjectAsync(
+            LatestReleaseByProjectQuery query)
+        {
+            var dbResult =
+                await _bizConfigObjectRepository.GetLatestReleaseConfigByProjectAsync(query.Projects,
+                    query.EnvClusterId);
+
+
+            TypeAdapterConfig<(int projectId, ConfigObjectRelease release), LatestReleaseConfigModel>.NewConfig()
+                .Map(dest => dest.ConfigObjectId, src => src.release.ConfigObjectId)
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Application/App/QueryHandler.cs:71]: release lowercase?

Comment (xyfy) [reply src/Services/Masa.Dcc.Service/Application/App/QueryHandler.cs:71]: it is paramName,not property

File: src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseByProjectQuery.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Dcc.Service.Admin.Application.App.Queries;
+
+public record LatestReleaseByProjectQuery(List<BuildingBlocks.StackSdks.Pm.Model.ProjectModel> Projects, int? EnvClusterId = null) : Query<List<LatestReleaseConfigModel>>
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseByProjectQuery.cs:6]: ProjectLatestReleaseQuery

File: src/Services/Masa.Dcc.Service/Domain/App/Repositories/IAppConfigObjectRepository.cs [hunk 5]
```diff
@@ -8,5 +8,8 @@ public interface IAppConfigObjectRepository : IRepository<AppConfigObject>
         Task<List<AppConfigObject>> GetListByAppIdAsync(int appId);
 
         Task<List<AppConfigObject>> GetListByEnvClusterIdAsync(int envClusterId, int appId);
+
+        Task<List<(int appId, ConfigObjectRelease release)>> GetLatestReleaseConfigByAppAsync(IEnumerable<int> appIds,
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Domain/App/Repositories/IAppConfigObjectRepository.cs:12]: GetAppLatestReleaseConfigAsync

File: src/Services/Masa.Dcc.Service/Domain/App/Repositories/IBizConfigObjectRepository.cs [hunk 6]
```diff
@@ -8,5 +8,8 @@ public interface IBizConfigObjectRepository : IRepository<BizConfigObject>
         Task<List<BizConfigObject>> GetListByEnvClusterIdAsync(int envClusterId, int bizConfigId);
 
         Task<List<BizConfigObject>> GetListByBizConfigIdAsync(int bizConfigId);
+
+        Task<List<(int projectId, ConfigObjectRelease release)>> GetLatestReleaseConfigByProjectAsync(
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Domain/App/Repositories/IBizConfigObjectRepository.cs:12]: GetProjectLatestReleaseConfigAsync

File: src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/AppConfigObjectRepository.cs [hunk 7]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Mvc.Formatters;
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/AppConfigObjectRepository.cs:4]: global using

File: src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/AppConfigObjectRepository.cs [hunk 8]
```diff
@@ -19,6 +21,40 @@ public async Task<List<AppConfigObject>> GetListByEnvClusterIdAsync(int envClust
...
+            if (appIds?.Any() != true)
+            {
+                return result;
+            }
+            var bizConfigs = Context.Set<AppConfigObject>()
+                .Where(x => appIds.Contains(x.AppId))
+                .Select(b => new { b.ConfigObjectId, b.AppId });
+            if (envClusterId.HasValue)
+            {
+                bizConfigs = Context.Set<AppConfigObject>()
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/AppConfigObjectRepository.cs:36]:
```
Expression<Func<AppConfigObject, bool>> condition = appConfig=> true;
if(envClusterId.HasValue){
     condition.And(XXXX);
}
```

File: src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/BizConfigObjectRepository.cs [hunk 9]
```diff
@@ -28,5 +31,63 @@ public async Task<List<BizConfigObject>> GetListByBizConfigIdAsync(int bizConfig
...
+            // TODO Group by is not support to join,it support in EFCore 7.0 https://github.com/dotnet/efcore/issues/24106
+
+
+            var qBizConfigs = Context.Set<BizConfigObject>()
+                   .Include(x => x.BizConfig)
+                   .Where(x => identities.Contains(x.BizConfig.Identity))
+                   .Select(b => new { b.BizConfig.Identity, b.ConfigObjectId });
+            if (envClusterId.HasValue)
+            {
+                qBizConfigs = Context.Set<BizConfigObject>().AsNoTracking()
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Infrastructure/Repositories/App/BizConfigObjectRepository.cs:67]: Expression

File: src/Services/Masa.Dcc.Service/Services/AppService.cs [hunk 10]
```diff
@@ -1,29 +1,43 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+using Masa.BuildingBlocks.StackSdks.Auth;
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Services/AppService.cs:3]: global using

File: src/Services/Masa.Dcc.Service/Services/ProjectService.cs [hunk 11]
```diff
@@ -25,21 +25,21 @@ public async Task<List<ProjectAppsModel>> GetProjectListAsync(string envName)
...
+    public async Task<ProjectDetailModel> GetAsync(int id)
     {
-        var result = await _pmClient.ProjectService.GetAsync(Id);
+        var result = await _pmClient.ProjectService.GetAsync(id);
 
         return result;
     }
 
-    public async Task<List<ProjectModel>> GetListAsync()
+    public async Task<List<BuildingBlocks.StackSdks.Pm.Model.ProjectModel>> GetListAsync()
```
Comment (MayueCif) [inline src/Services/Masa.Dcc.Service/Services/ProjectService.cs:35]: BuildingBlocks.StackSdks.Pm.Model.  ?

**Review by MayueCif — COMMENTED** [review]
@xyfy

File: src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs (hunk 1 above)
Comment (MayueCif) [reply src/Services/Masa.Dcc.Service/Application/App/Queries/LatestReleaseQueryByApp.cs]: AppLatestReleaseQuery better

File: src/Services/Masa.Dcc.Service/Services/ProjectService.cs (hunk 11 above)
Comment (xyfy) [reply src/Services/Masa.Dcc.Service/Services/ProjectService.cs:35]: it is confict with BuildingBlocks.StackSdks.Auth.Model.

Comment (xyfy) [reply src/Services/Masa.Dcc.Service/Services/ProjectService.cs:35]: image

**Review by MayueCif — APPROVED** [review]
LGTM

**Pull request merged**